tests/testthat/test-redundant_equals_linter.R

```diff
@@ -0,0 +1,30 @@
+test_that("redundant_equals_linter skips allowed usages", {
+  # comparisons to non-logical constants
+  expect_lint("x == 1", NULL, redundant_equals_linter())
+  # comparison to TRUE as a string
+  expect_lint("x != 'TRUE'", NULL, redundant_equals_linter())
+})
+
+test_that("redundant_equals_linter blocks simple disallowed usages", {
+  linter <- redundant_equals_linter()
+  eq_msg <- rex::rex("Using == on a logical vector is redundant.")
+  ne_msg <- rex::rex("Using != on a logical vector is redundant.")
+  expect_lint("x == TRUE", eq_msg, linter)
```

Comment:
These are definitely edge cases, but should these lint?
``` r
library(lintr)

lint(
  text = "x == c(TRUE)", 
  linters = redundant_equals_linter()
)

lint(
  text = "x != c(TRUE)", 
  linters = redundant_equals_linter()
)
```

<sup>Created on 2022-09-25 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I am fine with this being covered in a future PR. Up to you.

Reply:
I don't think it's worth adding :)

those will lint with unneeded_concatenation_linter(), and I suspect it's pretty uncommon to begin with

Reply:
Fair enough. 

I was just imagining a scenario where the user has turned off `unneeded_concatenation_linter()`, but the combined probability of both happening is probably minuscule.

Reply:
It's definitely a good thing to keep in mind -- we shouldn't rely on ppl having certain combinations of linters applied to get max value from our linters. It helps that `unneeded_concatenation_linter()` is on by default, so it's not like users have to go searching for 2 custom linters for that case.

But there is also a maintenance overhead tradeoff, and in this case I just don't see the edge case as worthwhile. I just searched our code history (20 years of commits) and nobody ever used `== c(TRUE)` (ditto `FALSE` and `!=`).